### DISCUSSION TITLE
[TemporaryUploadedFile] $tmpFile = tmpfile() returns false and throws error (line 23)

### DISCUSSION THREAD

**Discussion by valtermedeiros** [livewire/livewire#6439] [7. Bugs (archived)] [answered]
**Location**: src/Features/SupportFileUploads/TemporaryUploadedFile
```php
 public function __construct($path, $disk)
    {
        $this->disk = $disk;
        $this->storage = Storage::disk($this->disk);
        $this->path = FileUploadConfiguration::path($path, false);

        $tmpFile = tmpfile(); //<---------------------------------------------------------- THIS HERE : line 23

        parent::__construct(stream_get_meta_data($tmpFile)['uri'], $this->path);
    }
```

### Problem 
When "tmpfile()" on line 23 **returns false**, everything collapses, throwing this error:
stream_get_meta_data(): Argument #1 ($stream) must be of type resource, bool given

### Cause
tmpfile() returns false, and since it is a param after on stream_get_meta_data(), it fails and throws the error

### My fix was to use the Storage facade: 
$tmpFile = fopen(Storage::path($this->path),'r');
or
$tmpFile = fopen(storage()->path($this->path),'r');

```php
 public function __construct($path, $disk)
    {
        $this->disk = $disk;
        $this->storage = Storage::disk($this->disk);
        $this->path = FileUploadConfiguration::path($path, false);

        $tmpFile = fopen(Storage::path($this->path),'r'); //<---------------------------------------- FIXED

        parent::__construct(stream_get_meta_data($tmpFile)['uri'], $this->path);
    }
```

**Comment by PhiloNL** [comment, marked as answer]
Have you looked into why `tmpfile()` is returning `false`? It seems like a permission issue. Can you make sure your tmp folder is writable? You can find the folder by running `sys_get_temp_dir()`

**Reply by valtermedeiros** [reply]
Thanks. You are correct. I can confirm that the issue lies on the permissions of the "sys_get_temp_dir()" folder.

More info:
- That folder is defined on php.ini (run by your webserver) and since I'm running the webserver on MacOS it was pointing to a folder with no permissions to read/write/exec
- Modified the temp folder for a custom one, when using the webserver and is now solved.
